cmd/openshift-install/create.go

```diff
@@ -278,7 +284,67 @@ func destroyBootstrap(ctx context.Context, config *rest.Config, directory string
 	return destroybootstrap.Destroy(rootOpts.dir)
 }
 
-// waitForconsole returns the console URL from the route 'console' in namespace openshift-console
+// waitForInitializedCluster polls ClusterVersion waiting for confirmation that
+// the cluster has been initialized.
+func waitForInitializedCluster(ctx context.Context, config *rest.Config) error {
+	timeout := 30 * time.Minute
+	logrus.Infof("Waiting up to %v for the cluster to initialize...", timeout)
+	cc, err := configclient.NewForConfig(config)
+	if err != nil {
+		return errors.Wrap(err, "failed to create a config client")
+	}
+	clusterVersionContext, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+	// Poll quickly but only log when we've seen 15 of the same errors or output
+	// of no route in a row (to show we're still alive).
+	logDownsample := 15
+	silenceRemaining := logDownsample
+	lastLoggedMsg := ""
+	logMessage := func(msg string) {
+		if msg == lastLoggedMsg {
+			silenceRemaining--
+			if silenceRemaining > 0 {
+				return
+			}
+		}
+		logrus.Debugf(msg)
+		silenceRemaining = logDownsample
+		lastLoggedMsg = msg
+	}
+	var failureErr error
+	wait.Until(func() {
+		cv, err := cc.ConfigV1().ClusterVersions().Get("version", metav1.GetOptions{})
+		if err != nil {
+			logMessage(fmt.Sprintf("Still waiting for the ClusterVersion: %v", err))
+			return
+		}
+		for _, cond := range cv.Status.Conditions {
```

Comment:
Pull this out into a “findStatusCondition” instead, makes it much easier to read and reason about.

Reply:
similar to https://github.com/openshift/library-go/blob/86f3eb4ba0e678146cf47f61010c34c638cea2d2/pkg/config/clusteroperator/v1helpers/status.go#L53-L62